<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTextEdit" name="dimX">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>10</y>
      <width>161</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="dimY">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>50</y>
      <width>161</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="dimZ">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>90</y>
      <width>161</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>66</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>DIM X</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>60</y>
      <width>66</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>DIM Y</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>100</y>
      <width>66</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>DIM Z</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="iterations">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>130</y>
      <width>161</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>140</y>
      <width>151</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Number of iterations</string>
    </property>
   </widget>
   <widget class="QComboBox" name="generator">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>240</y>
      <width>86</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>CA</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>MC</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>250</y>
      <width>111</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Generator type</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>280</y>
      <width>151</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Neighbourhood type</string>
    </property>
   </widget>
   <widget class="QComboBox" name="neighbourhood">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>270</y>
      <width>86</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Moore</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Neumann</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Random</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>310</y>
      <width>131</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Nucleation</string>
    </property>
   </widget>
   <widget class="QComboBox" name="nucleation">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>300</y>
      <width>86</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Random</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Uniform</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="start">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>400</y>
      <width>141</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Run</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="current">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>340</y>
      <width>211</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Use current microstructure</string>
    </property>
   </widget>
   <widget class="QPushButton" name="reverse_grain">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>510</y>
      <width>221</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Reverse grains/pourus</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="remove_id">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>470</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="remove_grain">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>470</y>
      <width>151</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Remove grain (ID)</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="grain_number">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>170</y>
      <width>161</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>180</y>
      <width>151</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Number of grains</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>220</y>
      <width>131</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Boundry Condition</string>
    </property>
   </widget>
   <widget class="QComboBox" name="bc">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>210</y>
      <width>86</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Periodic</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Absorption</string>
     </property>
    </item>
   </widget>
   <widget class="QCheckBox" name="multithreading">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>370</y>
      <width>151</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Multithreading</string>
    </property>
   </widget>
   <widget class="QGraphicsView" name="result">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>10</y>
      <width>401</width>
      <height>411</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="decreaseZ">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>430</y>
      <width>111</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;</string>
    </property>
   </widget>
   <widget class="QPushButton" name="increaseZ">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>430</y>
      <width>111</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>&gt;</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="z_layer">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>430</y>
      <width>91</width>
      <height>23</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="export_2">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>510</y>
      <width>401</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Export to txt</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>start</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>OnStartClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>160</x>
     <y>422</y>
    </hint>
    <hint type="destinationlabel">
     <x>258</x>
     <y>413</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>current</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>OnCurrentChecked(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>21</x>
     <y>378</y>
    </hint>
    <hint type="destinationlabel">
     <x>291</x>
     <y>358</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>OnStartClick()</slot>
  <slot>OnCurrentChecked(int)</slot>
 </slots>
</ui>
